lib/ansible/modules/system/at.py

```diff
@@ -92,7 +92,10 @@ def add_job(module, result, at_cmd, count, units, command, script_file):
 
 def delete_job(module, result, at_cmd, command, script_file):
     for matching_job in get_matching_jobs(module, at_cmd, script_file):
-        at_command = "%s -d %s" % (at_cmd, matching_job)
+        if get_platform() != 'AIX':
```

Comment:
This should be an inclusive special case of `AIX` such as `== 'AIX'` instead of `!= 'AIX'`